packages/dart_code_builder/src/to_code.candy

```diff
@@ -0,0 +1,411 @@
+use ..nodes
+
+fun indent(code: String, indentFirstLine: Bool): String {
+  mut let s = ""
+  if indentFirstLine {
+    s = "  "
+  }
+  for char in code.characters() {
+    if char == "\n" {
+      s = "{s}\n  "
+    } else {
+      s = "{s}{char}"
+    }
+  }
+  s
+}
+
+fun parametersToCode(
+  required: Iterable<String>,
+  positional: Iterable<String>,
+  named: Iterable<String>,
+): String {
+  let required = required.join(", ")
+  let positional = positional.join(", ")
+  let named = named.join(", ")
+
+  mut let parameters = required
+  if !(required == "") && (!(positional == "") || !(named == "")) {
+    parameters = "{parameters}, "
+  }
+  if !(positional == "") {
+    parameters = "{parameters}[{positional}]"
+  }
+  if !(named == "") {
+    parameters = "{parameters}\👍{named}}"
+  }
+  parameters
+}
+
+fun annotationsToCode(annotations: Iterable<Annotation>): String {
+  (annotations as Iterable<Annotation>).map<String>({ "{(it as ToCode).toCode()}\n" }).join("")
+}
+
+trait ToCode {
+  fun toCode(): String
+}
+
+impl CompilationUnit: ToCode {
+  fun toCode(): String {
+    let directivesCode = (directives as Iterable<Directive>)
+      .map<String>({ (it as ToCode).toCode() })
+      .join("\n")
+    let declarationsCode = (declarations as Iterable<Declaration>)
+      .map<String>({ (it as ToCode).toCode() })
+      .join("\n")
+    "{directivesCode}\n\n{declarationsCode}"
+  }
+}
+
+impl Annotation: ToCode {
+  fun toCode(): String { "@{(expression as ToCode).toCode()}" }
+}
+
+impl Directive: ToCode
+
+impl Import: ToCode {
+  fun toCode(): String {
+    mut let s = "import '{path}'"
+    if prefix is Some {
+      s = "{s} as {prefix.unwrap()}"
+    }
+    if (show as Iterable).isNotEmpty() {
+      s = "{s} show {(show as Iterable).join(", ")}"
+    }
+    if (hide as Iterable).isNotEmpty() {
+      s = "{s} hide {(hide as Iterable).join(", ")}"
+    }
+    "{s};"
+  }
+}
+
+impl Part: ToCode {
+  fun toCode(): String { "part '{path}';" }
+}
+
+impl PartOf: ToCode {
+  fun toCode(): String { "part of '{path}';" }
+}
+
+let a = 3
```

Comment:
Does this have any purpose?

Reply:
Nope, just for debugging and seeing if the HIR gets available for type hints 😅